File: ios/Anemone SDK/general/extension/Coordinate+Json.swift
//
//  Coordinate+Json.swift
//  Anemone SDK
//
//  Created by Angel Luis Garcia on 16/06/2020.
//  Copyright © 2020 Syltek Solutions S.L. All rights reserved.
//

import Foundation

public extension Coordinate {
    static func from(json: JSONObject) throws -> Coordinate {
        Coordinate(
            lat: try json.getDouble("lat"),
            lon: try json.getDouble("lon")
        )
    }

    func toJson() -> JSONObject {
        let json = JSONObject()
        json.setDouble("lat", latitude)
        json.setDouble("lon", longitude)

        return json
    }
}

File: android/app/src/main/java/com/playtomic/test/anemone_sdk/general/extension/Coordinate+Json.kt
package com.anemonesdk.general.extension

import com.anemonesdk.general.json.JSONObject
import com.playtomic.foundation.model.Coordinate

fun Coordinate.Companion.from(json: JSONObject): Coordinate =
    Coordinate(
        lat = json.getDouble("lat"),
        lon = json.getDouble("lon")
    )

fun Coordinate.toJson(): JSONObject {
    val json = JSONObject()
    json.setDouble("lat", latitude)
    json.setDouble("lon", longitude)
    return json
}
